pkg/controller/upload_controller.go

```diff
@@ -128,28 +132,31 @@ func (c *uploadController) enqueueUploadItem(obj interface{}) {
 		return
 	}
 
-	log.Infof("Filtering out the upload request from nodes other than %v", c.nodeName)
-	peID, err := astrolabe.NewProtectedEntityIDFromString(req.Spec.SnapshotID)
-	if err != nil {
-		log.WithError(err).Errorf("Failed to extract volume ID from snapshot ID, %v", req.Spec.SnapshotID)
-		return
-	}
-
-	uploadNodeName, err := utils.RetrievePodNodesByVolumeId(peID.GetID())
-	if err != nil {
-		_, ok := err.(utils.NotFoundError)
-		if ok {
-			log.Infof("Trying to back independent PV from volume ID, %v", peID.String())
-			uploadNodeName = c.nodeName
-		} else {
-			log.WithError(err).Errorf("Failed to retrieve pod nodes from volume ID, %v", peID.String())
+	// If the data manager is local, check if current node is the expected upload node
```

Comment:
I think this comment is supposed to be about the case when data mgr is not local

Reply:
Updated the comment to
// Check if current node is the expected upload node only if data manager is not remote.